File: typescript/nth-prime/nth-prime.ts
export default class Prime {
    private static readonly start: number = 2

    nth(n: number): number {
        if (n < 1) {
            throw new Error("Prime is not possible")
        }

        const sieve: number[] = []
        let int: number = Prime.start
        while (sieve.length < n) {
            if (sieve.every((n) => int % n !== 0)) {
                sieve.push(int)
            }
            int++
        }

        return sieve[sieve.length - 1]
    }
}

File: scala/nth-prime/src/main/scala/NthPrime.scala
import scala.util.Try

object NthPrime {
  def prime(n: Int): Option[Int] = Try(sieve(2 to 1000000 toStream)(n - 1)).toOption

  private def sieve(numbers: Stream[Int], prior: Seq[Int] = Nil): Stream[Int] = {
    if (numbers.isEmpty) Stream.empty
    else {
      val number = numbers.head
      if (prior.forall(number % _ != 0)) number #:: sieve(numbers.tail, prior :+ number)
      else sieve(numbers.tail, prior)
    }
  }
}
